sdk/core/core-client/src/utils.ts

```diff
@@ -176,6 +178,6 @@ export function flattenResponse(
     shouldWrapBody:
       expectedBodyTypeName !== "Composite" &&
       expectedBodyTypeName !== "Dictionary" &&
```

Comment:
perhaps we can push these checks now into `isPrimitiveBody` now that it takes the `expectedBodyTypeName` as an arg.

Reply:
Good point! Updated.